src/Agent.Worker/Build/TeeTFCommandManager.cs

```diff
@@ -169,16 +206,21 @@ private string FormatArguments(params string[] args)
                 }
 
                 // Add the arg.
-                arguments.Append(arg != null && arg.Contains(" ") ? $@" ""{arg}""" : $" {arg}");
+                formattedArgs.Add(arg != null && arg.Contains(" ") ? $@"""{arg}""" : $"{arg}");
             }
 
-            // Add the collection/auth parameters.
-            arguments.Append($" -collection:{Endpoint.Url.AbsoluteUri} -login:_,{accessToken} -noprompt");
+            // Add the common parameters.
+            if (!formatFlags.HasFlag(FormatFlags.OmitCollectionUrl))
```

Comment:
One of the commands blows up if the collection url is specified.

I considered different options for passing in a flag. E.g. an optional boolean parameter, but due to params needing to go last and optional params needing to go last, I think it wouldn't work. I also considered just a regular bool at the front of the method signature, but in that case would want to consume it using the named parameter syntax, which again was a no-go since once you start a named parameter all following params need to use the named syntax as well.